src/app/services/product.service.ts

```diff
@@ -39,17 +40,27 @@ export class ProductService {
       .pipe(map((p) => p as Product));
   }
 
-  insert(product: Product): Observable<any> {
 
-    return this.http.post<Product>(`${environment.apiBase}/product`, product, this.headers)
-      .pipe(
-        map((response) => response)
-      );
+  insert(sendObj: Product): Observable<any> {
+
+      return this.http.post(`${environment.apiBase}/product`, sendObj ).pipe(map((response: any) => response));
   }
 
-  update(productName: string, productPrice: number, salePrice: number, prodStatus: string, id: number, interestThreshold: number, image: string): Observable<any> {
-    return this.http.put(`${environment.apiBase}/product`,
-      { name: productName, price: productPrice, salePrice: salePrice, status: prodStatus, productId: id, interestThreshold: interestThreshold, imageUrl: image })
+  update(productName: string, productPrice: number, salePrice: number,
```

Comment:
where is product id? why not just have the method accept a Product?

Reply:
verbosity reduced